servers/republik/script/prolong/paperInvoice.js

```diff
@@ -29,60 +30,162 @@ const log = printIds
 
 log('Start')
 
+const PROLONG_PACKAGE = '6b8897bf-7ab4-433c-92a1-64fafcd54417'
+const PROLONG_PACAKGEOPTION_ABO = {
+  id: 'c1cb2b16-8f03-4eec-851c-cea4204d7ff7',
+  price: 24000
+}
+const PROLONG_PACAKGEOPTION_BENEFACTOR = {
+  id: '35be2bea-4d3f-44e8-a224-2d17de6d6d51',
+  price: 100000
+}
+
+const dueDate = moment().add(30, 'days')
+
+const createPledgeAndPayments = async ({ transaction, option }, rows) =>
+  Promise.map(rows, async row => {
+    if (dry) {
+      return { ...row, pledge: {}, pledgeOption: {}, payment: {}, pledgePayments: {} }
+    }
+
+    const pledge = await transaction.public.pledges.insertAndGet({
+      packageId: PROLONG_PACKAGE,
+      userId: row.user.id,
+      status: 'WAITING_FOR_PAYMENT',
+      total: option.price,
+      donation: 0,
+      sendConfirmMail: false
+    })
+
+    const pledgeOption = await transaction.public.pledgeOptions.insertAndGet({
+      templateId: option.id,
+      pledgeId: pledge.id,
+      amount: 1,
+      price: option.price,
+      membershipId: row.membershipId
+    })
+
+    const payment = await transaction.public.payments.insertAndGet({
+      type: 'PLEDGE',
+      method: 'PAYMENTSLIP',
+      paperInvoice: true,
+      total: option.price,
+      status: 'WAITING',
+      dueDate
+    })
+
+    const pledgePayments = transaction.public.pledgePayments.insertAndGet({
+      pledgeId: pledge.id,
+      paymentId: payment.id,
+      paymentType: 'PLEDGE'
+    })
+
+    const cache = createCache({ prefix: `User:${row.user.id}` })
+    await cache.invalidate()
+
+    return { ...row, pledge, pledgeOption, payment, pledgePayments }
+  })
+
+const mapToCsv = d => ({
+  id: d.user.id,
+  user: d.user.name,
+  name: d.address.name,
+  line1: d.address.line1,
+  line2: d.address.line2,
+  postalCode: d.address.postalCode,
+  city: d.address.city,
+  country: d.address.country,
+  membershipSequenceNumber: d.membershipSequenceNumber,
+  hrid: d.hrid,
+  prolongBeforeDate: d.prolongBeforeDate,
+  admin: `https://admin.republik.ch/users/${d.user.id}`
+})
+
 PgDb.connect().then(async pgdb => {
   if (dry) {
     log("dry run: this won't change anything")
   }
 
-  const enrichWithProlongAndAddress = async users => Promise.map(
-    users,
-    async (user) => {
-      const prolongBeforeDate = await getProlongBeforeDate(user, {}, { pgdb, user: me })
-      if (prolongBeforeDate && moment(prolongBeforeDate).isBefore(PROLONG_BEFORE_DATE)) {
+  const enrichWithProlongAndAddress = async data => Promise.map(
+    data,
+    async (d) => {
+      const user = transformUser(d)
+      const prolongBeforeDate = await getProlongBeforeDate(
+        user, {}, { pgdb, user: me }
+      )
+      if (
+        prolongBeforeDate &&
+        moment(prolongBeforeDate).isBefore(PROLONG_BEFORE_DATE)
+      ) {
         const address = await pgdb.public.addresses.findOne({
-          id: user._raw.addressId
+          id: user._raw.addressId,
+          country: ['Schweiz', 'schweiz', 'Liechtenstein']
```

Comment:
@patte Variations in database found thus far:

```
CH
ch
schweiz
SCHWEIZ
Schweiz
Schweiz 
Schweiz / Greece
Suisse
Svizzera
Swiss
Switzerland
Liechtenstein
Liechtenstein 
```

via `SELECT "country" FROM "addresses" GROUP BY "country"`

Reply:
yes, but for this script, it's enough.